qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,122 @@
+            if index is not None:
+                circuit_index = index
+            else:
+                # if the given circuit is  a new one, make base parameter values for + and - epsilon
+                circuit_index = len(self._circuits)
+                self._circuit_ids[id(circuit)] = circuit_index
+                self._base_parameter_values_dict[
+                    circuit_index
+                ] = make_fin_diff_base_parameter_values(circuit, self._epsilon)
+                self._circuits.append(circuit)
+
+            circuit_parameters = self._circuits[circuit_index].parameters
+            base_parameter_values_list = []
+            gradient_parameter_values = np.zeros(len(circuit_parameters))
+
+            # a parameter set for the partial option
+            parameters = partial_ or circuit_parameters
+            param_set = set(parameters)
+
+            result_index = 0
+            result_index_map = {}
+            # bring the base parameter values for parameters only in the partial parameter set.
+            for i, param in enumerate(circuit_parameters):
+                gradient_parameter_values[i] = parameter_values_[i]
+                if param in param_set:
+                    base_parameter_values_list.append(
+                        self._base_parameter_values_dict[circuit_index][i * 2]
+                    )
+                    base_parameter_values_list.append(
+                        self._base_parameter_values_dict[circuit_index][i * 2 + 1]
+                    )
+                    result_index_map[param] = result_index
+                    result_index += 1
+            # add the given parameter values and the base parameter values
+            gradient_parameter_values_list = [
+                gradient_parameter_values + base_parameter_values
+                for base_parameter_values in base_parameter_values_list
+            ]
+            gradient_circuits = [self._circuits[circuit_index]] * len(
+                gradient_parameter_values_list
+            )
+            observable_list = [observable] * len(gradient_parameter_values_list)
+
+            job = self._estimator.run(
+                gradient_circuits, observable_list, gradient_parameter_values_list, **run_options
+            )
+            results = job.result()
+
+            # Combines the results and coefficients to reconstruct the gradient
+            # for the original circuit parameters
+            values = np.zeros(len(parameter_values_))
+            for i, param in enumerate(circuit_parameters):
+                if param not in param_set:
+                    continue
+                # plus
+                values[i] += results.values[result_index_map[param] * 2] / (2 * self._epsilon)
+                # minus
+                values[i] -= results.values[result_index_map[param] * 2 + 1] / (2 * self._epsilon)
+
+            gradients.append(values)
+            status.append(job.status())
```

Comment:
If the method is now fully synchronous, we would not care about the status anymore, right? (connected to Hamamura-san's comment on the `EstimatorGradientResult` implementation. At this point, do we need a dedicated result class? or could you just return the gradients? (applies to all classes)

Reply:
That's true. I will remove the status from everywhere.